### DISCUSSION TITLE
Pyarmor Generate without pip install

### DISCUSSION THREAD

**Discussion by VelazcoJD** [dashingsoft/pyarmor#1721] [Q&A] [answered]
I am using a self-contained version of Python, and I cannot install Pyarmor directly with Pip. I tried to obfuscate using the source code in this repository, but I do not see the same options in `pyarmor.py` as in `pyarmor.exe`. In pyarmor.exe, I see the `gen` option, but in pyarmor.py I see the `obfuscate` option instead (along other different ones). The obfuscate option outputs different files as compared to pyarmor.exe. I also don't see how one can activate the license directly with pyarmor.py. Is there a way to use the source code to generate the same way pyarmor.exe does? Or, can I compile to generate the pyarmor.exe from source using the self-contained version of Python I am using (which is based on 3.10.13)? This is for a Windows build. Thanks in advance.

**Comment by jondy** [comment, marked as answer]
You should check module `pyarmor.cli.__main__`, not `pyarmor.pyarmor`

pyarmor 8 equals `python -m pyarmor.cli`

**Reply by VelazcoJD** [reply]
Thanks! That's what I was looking for :)
